File: src/Core/Language/Translations/ES.hs
module Core.Language.Translations.ES
where

--------

space :: Char
space = ' '
-- Core
define,show' :: String
define  = "define"
show'   = "muestra"
-- Base
alfred,byalfred :: String
alfred      = "alfred"
byalfred    = "adios" ++ space : alfred
-- IO
write,show_type,show_value,show_variable :: String
write           = "escribe"
show_type       = show' ++ space : "el tipo de"
show_value      = show' ++ space : "el valor de"
show_variable   = show' ++ space : "la variable"
-- Memories
define_moment,goto :: String
goto            = "vete a"
define_moment   = define ++ space : "este momento como"
-- Standard
define_variable :: String
define_variable = define ++ space : "la variable"
-- Reserved words
reserved_as,reserved_equals :: String
reserved_as      = "como"
reserved_equals  = "igual a"
-- Errors
error_parse :: String
error_parse = "Error al analizar"

File: src/Core/Language/Translations/ES.py
# -*- coding: utf-8 -*-

Language = {
    'Invoke': 'alfred',
    'Syntax': {
        'a': {
            'adi[oó]s alfred': 'Base.Exit'
        },
        'c': {
            'crea un servidor echo(?: en el puerto (?P<port>\d+))?': 'Network.ECHO.Server',
            'crea un servidor discard(?: en el puerto (?P<port>\d+))?': 'Network.Discard.Server',
            'crea un servidor (?:web|http)(?: en el puerto (?P<port>\d+))?': 'Network.HTTP.Server'
        },
        'd': {
            'define la variable (?P<name>.+) como (?P<value>.+)': 'Standard.Define',
            'divide (?P<name>.+) entre (?P<number>\d+)': 'Math.Division'
        },
        'e': {
            'escribe "(?P<text>.+)"': 'IO.Write'
        },
        'm': {
            'muestra la variable (?P<name>.+)': 'IO.Show.Variable',
            'muestra el tipo de la variable (?P<name>.+)': 'IO.Show.Type',
            'muestra el valor de la variable (?P<name>.+)': 'IO.Show.Variable',
            'multiplica (?P<name>.+) por (?P<number>\d+)': 'Math.Multiplication'
        },
        'r': {
            'r[eé]stale (?P<number>\d+) a (?P<name>.+)': 'Math.Subtraction'
        },
        's': {
            's[uú]male (?P<number>\d+) a (?P<name>.+)': 'Math.Addition'
        },
        'v': {
            'vete al momento (?P<name>.+)': 'Control.Moments.Jump'
        },
        '?': ['define este momento como (?P<name>.+)']
    }
}
